# PhylogeneticFactorAnalysis
Public repository for TODO: paper citation

## Aquilegia pollination
### Raw data
- __Aquilegia.nex__: Aquilegia tree and phenotypes. Data is from [Whithall & Hodges (2007)](https://doi.org/10.1038/nature05857). While the raw data was not published with the original study, it was publicly available via the [Bodega Bay Phylogenetics](http://treethinkers.org/) archived website: http://bodegaphylo.wikispot.org (accessed via the [internet archive](https://archive.org/)).
### Processed data
- __aquilegia.csv__: CSV file containing parsed data from __Aquilegia.nex__. Produced by running __aquilegia_process.jl__ script.
- __aquilegia_binary.csv__: Manually edited copy of __aquilegia.csv__ where the categorical "syndrome" variable has been replaced with the three binary variables "hummingbird", "bumblebee", and "hawk moth" using information from Figure 3 of [Whithall & Hodges (2007)](https://doi.org/10.1038/nature05857).
- __aquilegia_newick.txt__: Newick-formatted tree file. Produced by running __aquilegia_process.jl__ script.
### Scripts
- __aquilegia_process.jl__: Julia script that parses information in __Aquilegia.nex__ and outputs __aquilegia.csv__ and __aquilegia_newick.txt__.
- __aquilegia_pfa.xml__: Input file with instructions for [PhylogeneticFactorAnalysis.jl](https://github.com/gabehassler/PhylogeneticFactorAnalysis.jl) run.
### Results
- __aquilegiaBinary.zip__: Zipped folder storing output of [PhylogeneticFactorAnalysis.jl](https://github.com/gabehassler/PhylogeneticFactorAnalysis.jl) run.
## Mammalian life history
### Raw data
- __PanTHERIA_1-0_WR05_Aug2008.txt__: Mammalian trait data sourced from the PanTHERIA ecological database ([Jones et al., 2009](https://doi.org/10.1890/08-1494.1)). File downloaded from [https://esapubs.org/archive/ecol/E090/184/](https://esapubs.org/archive/ecol/E090/184/).
- __ele_1307_sm_sa1.tre__: Mammalian phylogeny from [Fritz et al. (2009)](https://doi.org/10.1111/j.1461-0248.2009.01307.x). Downloaded from [here](https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1111%2Fj.1461-0248.2009.01307.x&file=ELE_1307_sm_SA1.tre).
- __mammals_newick.txt__: the __mammalST_MSW05_bestDates__ tree manually copied from  __ele_1307_sm_sa1.tre__.
### Processed data
- __mammals.csv__: CSV file containing subset of __PanTHERIA_1-0_WR05_Aug2008.txt__ data. Produced by running __mammals_process.jl__.
- __mammals_classification.csv__: CSV file with mammalian taxonomic classification used in mammalian phylogeny figure. Produced by running __mammals_process.jl__.
- __mammals_newick_processed.txt__: Newick-formatted tree file with relevat subset of taxa in __mammals_newick.txt__. Produced by running __mammals_process.jl__.
### Scripts
- __mammals_plots.jl__: Julia script for producing near publication-ready mammals plots. Outputs __mammalsLoadings.pdf__ and __mammals_factors.svg__. __mammals_factors.svg__ was manualled editted prior to inclusion in the manuscript. Relies on unzipped __mammals.zip__ contents, __mammals_labels.csv__, __mammals_classification.csv__, and __mammals_newick_processed.txt__ as inputs.
- __mammals_process.jl__: Julia script with inputs __PanTHERIA_1-0_WR05_Aug2008.txt__ and __mammals_newick.txt__ and outputs __mammals.csv__, __mammals_classification.csv__, and __mammals_newick_processed.txt__.
- __mammals_pfa.xml__: Input file with instructions for [PhylogeneticFactorAnalysis.jl](https://github.com/gabehassler/PhylogeneticFactorAnalysis.jl) run.
### Results
- __mammals.zip__: Zipped folder storing output of [PhylogeneticFactorAnalysis.jl](https://github.com/gabehassler/PhylogeneticFactorAnalysis.jl) run. Final BEAST log files are omitted due to size constraints, but can be generated by re-running __mammals_pfa.xml__ with [PhylogeneticFactorAnalysis.jl](https://github.com/gabehassler/PhylogeneticFactorAnalysis.jl).

## New World monkey cranial morphology
### Raw data
- __nwn_brain_coordinates.tps__: 3-d procrusted coordinates for New World monkey cranial landmarks and semi-landmarks. Same data as used in [Aristide et al. (2016)](https://doi.org/10.1073/pnas.1514473113), but not publicly available (contact us for file).
- __nwm_newick.txt__: Newick format phylogenetic tree for the New World monkeys. Sourced from Dataset_S02 in Supporting Information of [Aristide et al. (2016)](https://doi.org/10.1073/pnas.1514473113).
### Processed data
- __nwm_coord_means.csv__: Mean values for each cranial landmark accross all species (contact us for file).
- __nwm_coordinates.csv__: Coordinates processed for PFA. All values were centered to have 0 mean. Values were scaled by the __TODO__ (contact us for file)
- __nwm_newick_processed.csv__: Newick format phylogenetic tree with some taxa removed.



## Yeast domestication
### Raw data
- __gallone_s5.xlsx__: industrial yeast phenotypic data (with metadata). Table S5 from [Gallone et al. (2016)](https://doi.org/10.1016/j.cell.2016.08.020).
- __gallone_s5_values.csv__: manually re-formatted yeast phenotype values from [Gallone et al. (2016)](https://doi.org/10.1016/j.cell.2016.08.020) Table S5.
- __gallone_s5_metadata.csv__: manually re-formatted yeast phenotype metadata from [Gallone et al. (2016)](https://doi.org/10.1016/j.cell.2016.08.020) Table S5.
### Processed data
- __yeast_continuous.csv__: Yeast data ready for use in PFA. Produced by __yeast_process.jl__.
- __yeast_labels.csv__: Yeast trait labels used for plotting. Produced by __yeast_process.jl__.
- __yeast_rrw.nwk__: Newick-formatted tree file generated from processing output of __yeast_rrw.xml__ BEAST run with [TreeAnnotator](https://beast.community/treeannotator).
### Scripts
- __yeast_pfa.xml__: Input file with instructions for [PhylogeneticFactorAnalysis.jl](https://github.com/gabehassler/PhylogeneticFactorAnalysis.jl) run.
- __yeast_process.jl__: Julia script that inputs __gallone_s5_values.csv__ and __gallone_s5_metadata.csv__ and outputs ...
- __yeast_rrw.xml__: BEAST xml input file for inferring yeast phylogeny. Sequence data originated from 2.8 MB super-alignment from [Gallone et al. (2016)](https://doi.org/10.1016/j.cell.2016.08.020) (file to large to inclue in this repository; please contact us for access)

## Computational efficiency comparison
### Scripts
- __timing.R__: R script to generate computational efficiency figure. Uses __results.csv__ as input.
- __timing_analysis.jl__: Processes results of timing comparison to make SI Table 1.
- __timing_basic.jl__: Performs simulation-based computational efficiency comparison. Outputs __results.csv__ file.
### Results
- __results.csv__: CSV file summarizing results of simulation study. Produced by __timing_basic.jl__.


